web/src/popover_menus.js

```diff
@@ -238,6 +241,29 @@ export function show_schedule_confirm_button(send_at_time, not_from_flatpickr =
     $("#compose-send-button").hide();
 }
 
+export function show_schedule_notify_above_composebox(send_at_time) {
+    // If no time was selected, do not show the banner
+    // As called, this function will use the `selected_send_later_time`
+    // value set by `show_schedule_confirm_button()`
+    if (!send_at_time) {
+        return;
+    }
+    notifications.notify_above_composebox(
+        $t({defaultMessage: `Scheduling this message for {send_at_time}.`}, {send_at_time}),
+        "scheduled_message_banner",
+        null,
+        null,
+        null,
+    );
```

Comment:
We definitely want to use the `compose_banner` API directly -- the `notifications` wrapper for it is problematic because it probably has semantics for when to clear that we don't want.

Reply:
I drafted these changes for this, but it sounds like Alya and Vlad have a proposed design change for this that wouldn't require this banner, so I'm going to just drop this commit.

``` diff
diff --git a/web/src/compose_banner.ts b/web/src/compose_banner.ts
index 4bea736f47..d881626189 100644
--- a/web/src/compose_banner.ts
+++ b/web/src/compose_banner.ts
@@ -29,6 +29,7 @@ export const CLASSNAMES = {
     // unmute topic notifications are styled like warnings but have distinct behaviour
     unmute_topic_notification: "unmute_topic_notification warning-style",
     // warnings
+    scheduling_message: "scheduling_message",
     topic_resolved: "topic_resolved",
     recipient_not_subscribed: "recipient_not_subscribed",
     wildcard_warning: "wildcard_warning",
diff --git a/web/src/popover_menus.js b/web/src/popover_menus.js
index 3652bca647..ea2d69bbf5 100644
--- a/web/src/popover_menus.js
+++ b/web/src/popover_menus.js
@@ -9,6 +9,7 @@ import tippy, {delegate} from "tippy.js";
 
 import render_actions_popover_content from "../templates/actions_popover_content.hbs";
 import render_all_messages_sidebar_actions from "../templates/all_messages_sidebar_actions.hbs";
+import render_scheduling_message_banner from "../templates/compose_banner/scheduling_message.hbs";
 import render_compose_control_buttons_popover from "../templates/compose_control_buttons_popover.hbs";
 import render_compose_select_enter_behaviour_popover from "../templates/compose_select_enter_behaviour_popover.hbs";
 import render_delete_topic_modal from "../templates/confirm_dialog/confirm_delete_topic.hbs";
@@ -25,6 +26,7 @@ import * as channel from "./channel";
 import * as common from "./common";
 import * as compose from "./compose";
 import * as compose_actions from "./compose_actions";
+import * as compose_banner from "./compose_banner";
 import * as condense from "./condense";
 import * as confirm_dialog from "./confirm_dialog";
 import * as drafts from "./drafts";
@@ -248,9 +250,21 @@ export function show_schedule_confirm_button(send_at_time, not_from_flatpickr =
     $("#compose-send-button").hide();
 }
 
+function show_scheduling_message_banner() {
+    // TODO: Add a parameter to this for when we're editing a scheduled message.
+    const display_time = get_formatted_selected_send_later_time();
+    const rendered_banner = render_scheduling_message_banner({
+        banner_type: compose_banner.WARNING,
+        display_time,
+        classname: compose_banner.CLASSNAMES.scheduling_message,
+    });
+
+    compose_banner.append_compose_banner_to_banner_list(rendered_banner);
+}
 function update_scheduled_date_from_modal(send_at_time, not_from_flatpickr = false) {
     overlays.close_active_modal();
     show_schedule_confirm_button(send_at_time, not_from_flatpickr);
+    show_scheduling_message_banner();
 }
 
 export function initialize() {
diff --git a/web/templates/compose_banner/scheduling_message.hbs b/web/templates/compose_banner/scheduling_message.hbs
new file mode 100644
index 0000000000..166b6d555f
--- /dev/null
+++ b/web/templates/compose_banner/scheduling_message.hbs
@@ -0,0 +1,3 @@
+{{#> compose_banner }}
+    <p>{{#tr}}Scheduling this message for {display_time}.{{/tr}}</p>
+{{/compose_banner}}
```